tekton/ci/jobs.yaml

```diff
@@ -65,3 +65,65 @@ spec:
       echo "Content of the repo:"
       ls $(resources.inputs.source.path)
       exit $(( $(date +"%s") % 2 ))
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: build-push-kaniko
+spec:
+  params:
+  - name: IMAGE
+    description: Name (reference) of the image to build.
+  - name: DOCKERFILE
+    description: Path to the Dockerfile to build.
+    default: ./Dockerfile
+  - name: CONTEXT
+    description: The build context used by Kaniko.
+    default: ./
+  - name: EXTRA_ARGS
+    default: ""
+  - name: BUILDER_IMAGE
+    description: The image on which builds will run
+    default: gcr.io/kaniko-project/executor:latest
+  resources:
+    inputs:
+    - name: source
+      type: git
+    outputs:
+    - name: endtrigger
+      type: cloudEvent
+  results:
+  - name: IMAGE_DIGEST
+    description: Digest of the image just built.
+
+  steps:
+  - name: build-and-push
+    workingDir: /workspace/source
+    image: $(params.BUILDER_IMAGE)
+    # specifying DOCKER_CONFIG is required to allow kaniko to detect docker credential
+    # https://github.com/tektoncd/pipeline/pull/706
+    env:
+    - name: DOCKER_CONFIG
+      value: /tekton/home/.docker
+    command:
+    - /kaniko/executor
+    - $(params.EXTRA_ARGS)
+    - --dockerfile=$(params.DOCKERFILE)
+    - --context=$(workspaces.source.path)/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
+    - --destination=$(params.IMAGE)
+    - --oci-layout-path=$(workspaces.source.path)/image-digest
+    securityContext:
+      runAsUser: 0
+  - name: write-digest
+    workingDir: $(workspaces.source.path)
+    image: gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/imagedigestexporter:v0.11.1
+    # output of imagedigestexport [{"name":"image","digest":"sha256:eed29..660"}]
+    command: ["/ko-app/imagedigestexporter"]
+    args:
+    - -images=[{"name":"$(params.IMAGE)","type":"image","url":"$(params.IMAGE)","digest":"","OutputImageDir":"$(workspaces.source.path)/image-digest"}]
+    - -terminationMessagePath=image-digested
+  - name: digest-to-results
+    workingDir: $(workspaces.source.path)
+    image: stedolan/jq
+    script: |
+      cat image-digested | jq '.[0].digest' | tee /tekton/results/IMAGE_DIGEST
```

Comment:
Would passing `--digest-file /tekton/results/IMAGE_DIGEST` to kaniko achieve the same result here?
See https://github.com/GoogleContainerTools/kaniko/blob/master/README.md#--digest-file

Reply:
oh this is new :sweat_smile: :angel:

Reply:
so, you prefer to use the imagedigestexporter?

Reply:
as of this PR yes, but I'll prepare an update in `catalog` and will update this too :angel:

Reply:
NIT: It would be really useful if the last step logged the full URL of the IMAGE for people to easily copy/paste, but that's something we can add later